be/src/vec/exec/format/arrow/arrow_reader.cpp

```diff
@@ -0,0 +1,115 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow_reader.h"
+
+#include "arrow/array.h"
+#include "arrow/io/buffered.h"
+#include "arrow/io/stdio.h"
+#include "arrow/ipc/options.h"
+#include "arrow/ipc/reader.h"
+#include "arrow/record_batch.h"
+#include "arrow/result.h"
+#include "arrow_pip_input_stream.h"
+#include "common/logging.h"
+#include "io/fs/stream_load_pipe.h"
+#include "olap/wal_manager.h"
+#include "runtime/descriptors.h"
+#include "runtime/runtime_state.h"
+#include "vec/utils/arrow_column_to_doris_column.h"
+
+namespace doris {
+class RuntimeProfile;
+} // namespace doris
+
+namespace doris::vectorized {
+
+ArrowReader::ArrowReader(RuntimeState* state, RuntimeProfile* profile, ScannerCounter* counter,
+                         const TFileScanRangeParams& params, const TFileRangeDesc& range,
+                         const std::vector<SlotDescriptor*>& file_slot_descs, io::IOContext* io_ctx)
+        : _state(state), _range(range), _file_slot_descs(file_slot_descs), _file_reader(nullptr) {}
+
+ArrowReader::~ArrowReader() = default;
+
+Status ArrowReader::init_reader() {
+    RETURN_IF_ERROR(FileFactory::create_pipe_reader(_range.load_id, &_file_reader, _state, false));
+    _pip_stream = ArrowPipInputStream::create_unique(_file_reader);
+    return Status::OK();
+}
+
+Status ArrowReader::get_next_block(Block* block, size_t* read_rows, bool* eof) {
+    bool has_next = false;
+    RETURN_IF_ERROR(_pip_stream->HasNext(&has_next));
+    if (!has_next) {
+        *read_rows = 0;
+        *eof = true;
+        return Status::OK();
+    }
+
+    // create a reader to read data
+    arrow::Result<std::shared_ptr<arrow::ipc::RecordBatchStreamReader>> tRet =
+            arrow::ipc::RecordBatchStreamReader::Open(_pip_stream.get(),
+                                                      arrow::ipc::IpcReadOptions::Defaults());
+    if (!tRet.ok()) {
+        LOG(ERROR) << "failed to open stream reader: " << tRet.status().message();
+        return Status::InternalError("failed to open stream reader: {}", tRet.status().message());
+    }
+    auto reader = std::move(tRet).ValueUnsafe();
+
+    // get arrow data from reader
+    arrow::Result<arrow::RecordBatchVector> tRet2 = reader->ToRecordBatches();
+    if (!tRet2.ok()) {
+        LOG(ERROR) << "failed to read batch: " << tRet2.status().message();
+        return Status::InternalError("failed to read batch: {}", tRet2.status().message());
+    }
+    std::vector<std::shared_ptr<arrow::RecordBatch>> out_batches = std::move(tRet2).ValueUnsafe();
+
+    // convert arrow batch to block
+    cctz::time_zone ctzz;
+    TimezoneUtils::find_cctz_time_zone("UTC", ctzz);
```

Comment:
2 questions:
1. Why using UTC
2. If always be UTC, this should be done once when init the arrow reader

Reply:
Since the timestamp type is not yet supported, this is just a placeholder. This value will be modified later based on the schema information.
We can add a TODO here.